File: week7/methodoverload.cpp
#include <iostream>
using namespace std;
class MeOl{
    public:
    int add(int a, int b){
        return a+b;
    }
    double add(double a, double b){
        return a+b;
    }
    int add(int a, int b, int c){
        return a+b+c;
    }
};


int main(){
    
    MeOl obj;
    cout << "add(int 10, int 20) : " << obj.add(10,20) << endl;
    cout << "add(double 25.34, double 73.57) : " << obj.add(25.34,73.57) << endl;
    cout << "add(int 10, int 20, int 30) : " << obj.add(10,20,30) << endl;
    
}

File: week7/methodoverload.java
class A {
    public void mul(int a, int b) {
        System.out.println(a + " X " + b + " = " + (a * b));
    }

    public void mul(int a, int b, int c) {
        System.out.println(a + " X " + b + " X " + c + " = " + (a * b * c));
    }

    public void mul(double a, double b) {
        System.out.println(a + " X " + b + " = " + (a * b));
    }

    public void mul(double a, double b, double c) {
        System.out.println(a + " X " + b + " X " + c + " = " + (a * b * c));
    }
}

class methodOverloading {
    public static void main(String[] args) {
        A obj = new A();
        obj.mul(3, 4);
        obj.mul(3.4, 2.6);
        obj.mul(9.2, 2.1, 6.3);
    }
}
